conda/env/installers/conda.py

```diff
@@ -53,16 +96,78 @@ def install(prefix, specs, args, env, *_, **kwargs):
             update_modifier=UpdateModifier.FREEZE_INSTALLED,
         )
     except (UnsatisfiableError, SystemExit) as exc:
-        # See this comment for 'allow_retry' details
-        # https://github.com/conda/conda/blob/b4592e9eb0/conda/cli/install.py#L417-L429
-        if not getattr(exc, "allow_retry", True):
-            raise
-        unlink_link_transaction = solver.solve_for_transaction(
-            prune=getattr(args, "prune", False), update_modifier=NULL
+        # Handle solver errors (original code continues here)
+        raise exc
+
+    return _execute_transaction(unlink_link_transaction)
+
+
+def _install_explicit_environment(
+    prefix: str, specs: list, env: ExplicitEnvironment
+) -> dict:
+    """
+    Install packages from an explicit environment without using the solver.
+
+    Implements CEP-23 requirement to bypass the solver for explicit environments.
+
+    :param prefix: Installation target directory
+    :type prefix: str
+    :param specs: Package specifications to install
+    :type specs: list
+    :param env: Environment with explicit package URLs
+    :type env: ExplicitEnvironment
+    :return: Installation result from explicit()
+    :rtype: dict
+    """
+    from ...misc import explicit
+
+    log = logging.getLogger(__name__)
+
+    # Use verbose output if not in quiet mode
+    verbose = not getattr(context, "quiet", False)
+
+    # Determine which package specs to use (priority order):
+    explicit_specs = None
+    filename = env.explicit_filename
+
+    # 1. Try to read from original file if available (most reliable source)
+    if filename and os.path.exists(filename):
+        try:
+            with open(filename) as f:
+                explicit_specs = [
+                    line.strip()
+                    for line in f
+                    if line.strip() and not line.strip().startswith("#")
+                ]
+            log.debug(f"Using package specs from explicit file: {filename}")
+        except OSError as e:
+            log.warning(f"Could not read explicit file {filename}: {e}")
```

Comment:
This logic exists in two (or even three see conda.cli.spec_from_line) locations. Can this be de-duped or collected into a utility function. 
Fine leaving this as a future refactor.